pydicom/util/fixer.py

```diff
@@ -36,10 +35,16 @@ def fix_separator_callback(
     if try_replace:
         # Note value has not been decoded yet when this function called,
         #    so need to replace backslash as bytes
+        new_value = None
         if raw_elem.value is not None:
-            new_value = raw_elem.value.replace(
-                kwargs['invalid_separator'], b"\\"
-            )
+            if kwargs['invalid_separator'] == b" ":
+                new_value = raw_elem.value.strip().replace(
+                                kwargs['invalid_separator'], b"\\"
```

Comment:
A trailing space is usually there to avoid illegal odd value size - you probably should add a space after the replacement if the size is odd.

Reply:
See change where any stripped spaces are added to the end after the separator replacement. I tried to test this by writing the element out to a `DicomBytesIO` but it seems to already pad automatically to even.

However both this change, and the prior behavior, could result in odd value sizes if the invalid separator is more than one byte.

Reply:
> However both this change, and the prior behavior, could result in odd value sizes if the invalid separator is more than one byte.

Yes, I noticed this but found it unlikely that this will be used. However, I overlooked the fact that the padding will be done automatically later anyway, so this is probably not needed.